pkg/test/framework/environment/environment.go

```diff
@@ -214,6 +222,14 @@ type (
 		// The raw content of the response
 		Raw string
 	}
+
+	// DeployedCitadel represents a deployed Citadel instance.
+	DeployedCitadel interface {
```

Comment:
I believe the lifecycle of the secret should be tied to the lifecycle of this component (e.g. the component doesn't initialize until the secret exists), no? If so, I don't think you need these methods at all.

Reply:
The lifecycle might not be the same as lifecycle of Citadel. I am going to migrate another integrate test [TestCertificateRotation](https://github.com/istio/istio/blob/19315ed01763585525377460a7fa72237dc3107c/security/tests/integration/certificateRotationTest/certificate_rotation_test.go#L45) which configures Citadel to 
generate short life secrets [here](https://github.com/istio/istio/blob/19315ed01763585525377460a7fa72237dc3107c/security/tests/integration/istio_ca_cert_rotation_env.go#L87).